src/xbot/common/subsystems/pose/BasePoseSubsystem.java

```diff
@@ -0,0 +1,185 @@
+        inherentRioPitch = propManager.createPersistentProperty("InherentRioPitch", 0.0);
+        inherentRioRoll = propManager.createPersistentProperty("InherentRioRoll", 0.0);
+    }
+    
+    private void updateCurrentHeading() {
+        // Old heading - current heading gets the delta heading        
+        double imuDeltaYaw = lastImuHeading.difference(imu.getYaw());
+
+        // add the delta to our current
+        currentHeading.shiftValue(imuDeltaYaw);
+        
+        // update the "old" value
+        lastImuHeading = imu.getYaw();
+        
+        currentHeadingProp.set(currentHeading.getValue());
+        
+        currentPitch.set(getRobotPitch());
+        currentRoll.set(getRobotRoll());
+    }  
+    
+    private void updateDistanceTraveled(double currentLeftDistance, double currentRightDistance) {
+       
+        leftDriveDistance.set(currentLeftDistance);
+        rightDriveDistance.set(currentRightDistance);
+        
+        double deltaLeft = currentLeftDistance - previousLeftDistance;
+        double deltaRight = currentRightDistance - previousRightDistance;
+        
+        double totalDistance = (deltaLeft + deltaRight) / 2;
+        
+        // get X and Y
+        double deltaY = Math.sin(currentHeading.getValue() * Math.PI / 180) * totalDistance;
+        double deltaX = Math.cos(currentHeading.getValue() * Math.PI / 180) * totalDistance;
+        
+        totalDistanceX.set(totalDistanceX.get() + deltaX);
+        totalDistanceY.set(totalDistanceY.get() + deltaY);
+        
+        // save values for next round
+        previousLeftDistance = currentLeftDistance;
+        previousRightDistance = currentRightDistance;
+    }
+    
+    public ContiguousHeading getCurrentHeading() {
+        updateCurrentHeading();
+        return currentHeading.clone();
+    }
+    
+    public XYPair getFieldOrientedTotalDistanceTraveled() {
+        return getTravelVector().clone();
+    }
+    
+    private XYPair getTravelVector() {
+        return new XYPair(totalDistanceX.get(), totalDistanceY.get());
+    }
+    
+    public XYPair getRobotOrientedTotalDistanceTraveled() {
+        // if we are facing 90 degrees, no change.
+        // if we are facing 0 degrees (right), this rotates left by 90. Makes sense - if you rotate right, you want
+        // your perception of distance traveled to be that you have gone "leftward."
+        return getTravelVector().rotate(-(currentHeading.getValue() - 90)).clone();
+    }
+    
+    public void resetDistanceTraveled() {
+        totalDistanceX.set(0);
+        totalDistanceY.set(0);
+    }
+    
+    public void setCurrentHeading(double headingInDegrees){
+        currentHeading.setValue(headingInDegrees);
+    }
+    
+    /**
+     * This should be called as often as reasonably possible, to increase accuracy
+     * of the "distance traveled" calculation.
+     * 
+     * The PoseSubsystem can't directly own CANTalons, so some command will need to feed in the
```

Comment:
minor comment tweak, rather than discussing Talons here maybe it's just the drive positional sensors?